sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getInProgressApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,409 @@
+    await db.educationProgramOffering.save(futureOffering);
+
+    // Create an in-progress student assessment for the previous offering.
+    const futureAssessment = createFakeStudentAssessment(
+      {
+        auditUser: futureApplication.student.user,
+        application: futureApplication,
+        offering: futureOffering,
+      },
+      {
+        initialValue: {
+          calculationStartDate: addDays(70),
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+        },
+      },
+    );
+    await db.studentAssessment.save(futureAssessment);
+
+    // Create CRA income verifications for student.
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application: currentApplication,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    await db.craIncomeVerification.save([studentCRAIncomeVerification]);
+
+    // Mock assessment service to return the saved student assessment.
+    await mockOutstandingAssessment(
+      appModule,
+      currentApplication.currentAssessment,
+    );
+
+    const endpoint = `/students/application/${currentApplication.id}/in-progress`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    // The current assessment is not blocked by the future assessment because it is assessed first and
+    // the offering study start date is earlier than the future offering.
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: currentApplication.id,
+        applicationStatus: currentApplication.applicationStatus,
+        pirStatus: currentApplication.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        outstandingAssessmentStatus: SuccessWaitingStatus.Success,
+      });
+
+    // Change the Submitted and In-progress assessments to Completed to allow other test cases.
+    currentApplication.currentAssessment.studentAssessmentStatus =
+      StudentAssessmentStatus.Completed;
+    futureAssessment.studentAssessmentStatus =
+      StudentAssessmentStatus.Completed;
+    await db.studentAssessment.save([
+      currentApplication.currentAssessment,
+      futureAssessment,
+    ]);
+  });
+
+  it("Should get application in-progress details when there is an in-progress student assessment for an offering with earlier study start date.", async () => {
+    // Arrange
+
+    const currentApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+
+    // Create a previous application with an offering whose study start date and end date are much earlier
+    // than the current offering ones.
+    const previousApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+    const previousOffering = await db.educationProgramOffering.findOne({
```

Comment:
For this test, the following level of minimal data preparation must be sufficient. 

```ts
// Arrange
    // Application to get in-progress details.
    const currentApplication = await saveFakeApplication(
      db.dataSource,
      {
        student,
      },
      { applicationStatus: ApplicationStatus.InProgress },
    );

    // Create a previous application with an offering whose study start date and end date are much earlier
    // than the current offering ones.
    const previousApplication = await saveFakeApplication(
      db.dataSource,
      {
        student,
      },
      { applicationStatus: ApplicationStatus.InProgress },
    );
    const currentApplicationOffering =
      currentApplication.currentAssessment.offering;
    const previousApplicationOffering =
      previousApplication.currentAssessment.offering;
    previousApplicationOffering.studyStartDate = getISODateOnlyString(
      addDays(-10, currentApplicationOffering.studyStartDate),
    );
    await db.educationProgramOffering.save(previousApplicationOffering);
```